### PR TITLE
Fix/databaseinventory

### PR DESCRIPTION
**Pull request by trasher**
| Q             | A
| ------------- | ---
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Deprecations? | yes
| Tests pass?   | yes

### CODE REVIEW

File: inc/features/inventoriable.class.php [hunk 1]
```diff
@@ -134,26 +142,44 @@ protected function showInventoryInfo() {
       echo '</th>';
       echo '</tr>';
 
-      if (($agent = $this->getInventoryAgent()) === null) {
+      if ($agent === null) {
          echo '<tr class="tab_bg_1">';
          echo '<td colspan="4">'.__('No agent has been linked.').'</td>';
          echo "</tr>";
          return;
```
Comment (trasher) [inline inc/features/inventoriable.class.php:149]:
```suggestion
```

File: install/migrations/update_9.5.x_to_10.0.0/databases.php [hunk 2]
```diff
@@ -68,6 +68,21 @@
    $DB->queryOrDie($query, "10.0 add table glpi_databaseinstancecategoriess");
 }
 
+if ($DB->tableExists('glpi_databaseinstances_items')) {
```
Comment (cedric-anne) [inline install/migrations/update_9.5.x_to_10.0.0/databases.php:71]: It failed on my database.

```
PHP User Warning (512): Table glpi_databasesinstances does not exists in /var/www/glpi/inc/dbmysql.class.php at line 952

Warning: Table glpi_databasesinstances does not exists in /var/www/glpi/inc/dbmysql.class.php on line 952
PHP User Warning (512): Table glpi_databasesinstances does not exists in /var/www/glpi/inc/dbmysql.class.php at line 952

Warning: Table glpi_databasesinstances does not exists in /var/www/glpi/inc/dbmysql.class.php on line 952
PHP User Warning (512): Table item does not exists in /var/www/glpi/inc/dbutils.class.php at line 500

Warning: Table item does not exists in /var/www/glpi/inc/dbutils.class.php on line 500
SQL Error "1146": Table 'glpi.master.glpi_databasesinstances' doesn't exist in query "ALTER TABLE `glpi_databasesinstances` ADD `itemtype` VARCHAR(255) COLLATE utf8mb4_unicode_ci DEFAULT NULL   AFTER `states_id` , ADD `items_id` INT NOT NULL DEFAULT '0'   AFTER `itemtype` "

In dbmysql.class.php line 403:
                                                                                                                                                     
  10.0.0 multiple alter in glpi_databasesinstances - Error during the database query: ALTER TABLE `glpi_databasesinstances` ADD `itemtype` VARCHAR(  
  255) COLLATE utf8mb4_unicode_ci DEFAULT NULL   AFTER `states_id` ,                                                                                 
  ADD `items_id` INT NOT NULL DEFAULT '0'   AFTER `itemtype`  - Error is Table 'glpi.master.glpi_databasesinstances' doesn't exist                   
                                                                                                                                   
```

File: install/migrations/update_9.5.x_to_10.0.0/databases.php [hunk 3]
```diff
@@ -68,6 +68,21 @@
...
+   //1- migrate glpi_databasesinstances table
+   $migration->addField('glpi_databasesinstances', 'itemtype', 'string', [
+      'after' => 'states_id'
+   ]);
+   $migration->addField('glpi_databasesinstances', 'items_id', 'integer', [
+      'after' => 'itemtype'
+   ]);
+   $migration->addKey('item', ['itemtype', 'items_id']);
+   $migration->migrationOneTable('glpi_databases_instances');
```
Comment (cedric-anne) [inline install/migrations/update_9.5.x_to_10.0.0/databases.php:80]:
```suggestion
   $migration->migrationOneTable('glpi_databasesinstances');
```

File: inc/databaseinstance.class.php [hunk 4]
```diff
@@ -107,13 +107,65 @@ public function getDatabases(): array {
...
+            'name'                  => "items_id",
+            'width'                 => 'unset'
+         ];
+
+         Ajax::updateItemOnSelectEvent(
+            "dropdown_itemtype$rand",
+            "results_itemtype$rand",
+            $CFG_GLPI["root_doc"]."/ajax/dropdownAllItems.php", $p);
+
+            echo "<span id='results_itemtype$rand'>\n";
```
[the remark this answers is not in the document]
Comment (cedric-anne) [reply inc/databaseinstance.class.php]: fixed

File: install/migrations/update_9.5.x_to_10.0.0/databases.php (hunk 2 above)
Comment (cedric-anne) [reply install/migrations/update_9.5.x_to_10.0.0/databases.php]: fixed

File: install/migrations/update_9.5.x_to_10.0.0/databases.php (hunk 3 above)
Comment (cedric-anne) [reply install/migrations/update_9.5.x_to_10.0.0/databases.php]: fixed

**Review by cedric-anne — APPROVED** [review]

**Review by cedric-anne — APPROVED** [review]

**Review by orthagh — APPROVED** [review]

**Pull request merged**